### PR TITLE
Create GraphQL from the node schemas in dynamodb

### PR DESCRIPTION
**Pull request by wimax-grapl**
### Which issue does this PR correspond to?
grapl-security/issue-tracker#357

### How were these changes tested?
there's a decent amount of testing on this that exposes the issues.

I ran
`make test-integration TARGET=graphql-endpoint-tests`
and got the following from
`docker logs grapl-integration_tests_grapl-graphql-endpoint_1`
https://gist.github.com/wimax-grapl/812723d5c4843a8072406d4460c7f5d9 here

### CODE REVIEW

File: src/js/graphql_endpoint/modules/schema.ts [hunk 1]
```diff
@@ -1,255 +1,22 @@
...
-	name: "LensNode",
-	fields: () => ({
-		...BaseNode,
-		lens_name: { type: GraphQLString },
-		score: { type: GraphQLInt },
-		scope: { type: GraphQLList(GraplEntityType) },
-		lens_type: { type: GraphQLString },
-	}),
-});
-
```
Comment (wimax-grapl) [inline src/js/graphql_endpoint/modules/schema.ts:30]: This stuff is all now dynamically generated

File: src/js/graphql_endpoint/modules/root_query.ts [hunk 2]
```diff
@@ -1,372 +1,341 @@
...
-		console.debug("Lens response from DGraph", res);
-		return res.getJson()["all"];
-	} catch (e) {
-		console.error("Error in DGraph txn getLenses: ", e);
-	} finally {
-		console.debug("Closing Dgraph Txn in getLenses");
-		await txn.discard();
-	}
+    console.debug("first, offset parameters in getLenses()", first, offset);
+
```
Comment (wimax-grapl) [inline src/js/graphql_endpoint/modules/root_query.ts:30]: Unfortunately andrea and I got into a formatting war here. I'm gonna add a `make format` for TS asap.

File: src/js/graphql_endpoint/modules/root_query.ts [hunk 3]
```diff
@@ -1,372 +1,341 @@
...
+    uid: uidAsInt(node),
+    dgraph_type: dgraph_types,
+    display: display,
+  };
 }
 
-const AWS = require("aws-sdk");
-const IS_LOCAL = process.env.IS_LOCAL == "True" || null;
-
-const getDisplayProperty = async (nodeType: string) => {
```
Comment (wimax-grapl) [inline src/js/graphql_endpoint/modules/root_query.ts:151]: This is all replaced with the SchemaClient

File: src/js/graphql_endpoint/modules/schema.ts [hunk 4]
```diff
@@ -1,255 +1,22 @@
-import { GraphQLJSONObject } from "graphql-type-json";
-
```
Comment (wimax-grapl) [inline src/js/graphql_endpoint/modules/schema.ts:2]: # Don't forget to review the above file, root_query.ts

File: src/js/graphql_endpoint/tsconfig.json [hunk 5]
```diff
@@ -16,6 +16,7 @@
     "module": "commonjs",
     "moduleResolution": "node",
     "noImplicitAny": true,
+    // TODO: "strictNullChecks": true,
```
Comment (inickles-grapl) [inline src/js/graphql_endpoint/tsconfig.json:19]: Was this meant to be addressed in this PR?

File: src/js/graphql_endpoint/modules/schema_to_graphql.ts [hunk 6]
```diff
@@ -0,0 +1,168 @@
...
+}
+
+function getHardcodedOverride(
+    property: SchemaProperty,
+    typeMap: GqlTypeMap
+): GraphQLOutputType | undefined {
+    /* grapl_analyzerlib is *wrong*, and fixing it breaks other services.
+       so, for now, just manually override and fix it in a followup PR. 
+       context at:
+       https://grapl-internal.slack.com/archives/C017PLQ8TCZ/p1617991501057700
```
Comment (inickles-grapl) [inline src/js/graphql_endpoint/modules/schema_to_graphql.ts:68]: Sounds like we should have an issue in the tracker for this.

**Review by inickles-grapl — APPROVED** [review]
Just a couple comments, but not worth blocking on imo. 👍

File: src/js/graphql_endpoint/modules/schema_to_graphql.ts (hunk 6 above)
Comment (wimax-grapl) [reply src/js/graphql_endpoint/modules/schema_to_graphql.ts:68]: done https://github.com/grapl-security/issue-tracker/issues/388

File: src/js/graphql_endpoint/tsconfig.json (hunk 5 above)
Comment (wimax-grapl) [reply src/js/graphql_endpoint/tsconfig.json:19]: no, though i'll tackle it as part of my TODO project

**Pull request merged**